astroquery/mast/core.py

```diff
@@ -439,6 +439,14 @@ def _fabric_request(self, method, url, params=None, data=None, headers=None,
         response.raise_for_status()
         return [response]
 
+    def _request_w_cache(self, method, url, data=None, headers=None, retrieve_all=True,
```

Comment:
No need for this, _request() knows about cache

Reply:
Not using the `cache` in `BaseQuery._request()` is deliberate per note from @ceb . 

MAST service uses a long-running polling mechanism such that the raw http response from MAST is really just a status code, not the actual responses.

Reply:
ok, gotcha. Would you leave a comment about that somewhere in the code, please?

Reply:
It has been in the docstring of `MastClass._request()`. Please advice any if further clarification is needed.
 https://github.com/astropy/astroquery/blob/ef94db608663835a3d001c25423a58dcf8b6faf4/astroquery/mast/core.py#L442-L457